light/rpc/client.go

```diff
@@ -591,6 +592,19 @@ func (c *Client) ProveShares(
 	return res, err
 }
 
+// ProveSharesV2 calls rpcclient#ProveSharesV2 method and returns an NMT proof for a set
+// of shares, defined by `startShare` and `endShare`, to the corresponding rows.
+// Then, a binary merkle inclusion proof from the latter rows to the data root.
```

Comment:
nit: I find this go doc difficult to read.

> calls rpcclient#ProveSharesV2 method

that seems like implementation details that the caller doesn't need to know about.

> and returns an NMT proof for a set of shares, defined by `startShare` and `endShare`, to the corresponding rows.

It's not clear to me if one proof is returned or multiple proofs. I _think_ that this would return one NMT proof per row that is occupied by the share range [startShare, endShare). Additionally, I think the NMT proof proves that some shares exist in an NMT with a particular row root.

> // Then, a binary merkle inclusion proof from the latter rows to the data root.

The "latter rows" here refer to the rows that are occupied by the share range?

IMO this Go doc could be made clearer by stating something like:

```suggestion
// ProveSharesV2 returns a result share proof that includes multiple proofs to prove that a range of shares exist in a particular block. See ____ for docs on how to verify the share proof.
```

and then the `___` docs are located on a Go function like: `ResultShareProof.Proof.Verify()`.

Reply:
```suggestion
// ProveSharesV2 returns a share to data root inclusion proof.
```
Let's just simplify it like this.

Reply:
IMO that isn't clear because this proof is for a range of shares not an individual share.

Reply:
```suggestion
// ProveSharesV2 returns a shares to data root inclusion proof.
```
What do you think?

Reply:
idk why I still find that phrasing confusing. My alternative suggestion would be:

```
// ProveSharesV2 returns a proof of inclusion for a share range to the data root. Note: this proof is composed of multiple proofs.
```